windows/hcsshim.go

```diff
@@ -90,23 +90,33 @@ func newWindowsContainerConfig(ctx context.Context, owner, id string, spec *spec
 	var di = hcsshim.DriverInfo{
 		HomeDir: filepath.Dir(layerFolderPath),
 	}
-	conf.VolumePath, err = hcsshim.GetLayerMountPath(di, layerID)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to getmount path for layer %s: driverInfo: %#v", id, di)
-	}
 
 	if spec.Windows.HyperV != nil {
 		conf.HvPartition = true
-		for _, layerPath := range layerFolders {
-			utilityVMPath := spec.Windows.HyperV.UtilityVMPath
-			_, err := os.Stat(utilityVMPath)
-			if err == nil {
-				conf.HvRuntime = &hcsshim.HvRuntime{ImagePath: utilityVMPath}
-				break
-			} else if !os.IsNotExist(err) {
-				return nil, errors.Wrapf(err, "failed to access layer %s", layerPath)
+		var utilityPath string
+		if spec.Windows.HyperV.UtilityVMPath != "" {
+			if _, err := os.Stat(spec.Windows.HyperV.UtilityVMPath); err != nil {
+				return nil, errors.Wrapf(err, "stat %s", spec.Windows.HyperV.UtilityVMPath)
+			}
+			utilityPath = spec.Windows.HyperV.UtilityVMPath
+		} else {
+			for _, path := range layerFolders {
+				vmpath := filepath.Join(path, "UtilityVM")
+				if _, err := os.Stat(vmpath); err == nil {
```

Comment:
error out if the error is different than not exist?

Reply:
We still error out below if it's not available.  I think since we are iterating over layers looking for this thing, it would be better to not error out as we don't control what is in the layers.  the only other error that i can think of is a perm error and that is basically the same as not found so we should keep searching

Reply:
Yes, but we do lose the "permission denied" in that case, maybe log it as a warning at least?